src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -92,13 +119,54 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
 
-    async deinitialize(): Promise<InternalCheckoutSelectors> {
+    async deinitialize({gatewayId}: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer(gatewayId);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string, methodId?: any, paypalcommerce?: any ) {
+        if (gatewayId !== PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS)  {
+            return;
+        }
+
+        this._pollingInterval = window.setTimeout(async () => {
+            try {
+                this._pollingTimer += POLLING_INTERVAL;
+
+                const { status } = await this._paypalCommerceRequestSender.getOrderStatus();
+
+                if (status === ORDER_STATUS_APPROVED) {
+                    this._deinitializePollingTimer(gatewayId);
+                    this._tokenizePayment({ orderID: this._paypalCommercePaymentProcessor.getOrderId() }, submitForm);
+                } else if (status === ORDER_STATUS_CREATED  && this._pollingTimer < POLLING_MAX_TIME) {
+                    this._initializePollingMechanism(submitForm, gatewayId, methodId, paypalcommerce);
+                } else {
+                    this._reinitializeButtons({ gatewayId, methodId, paypalcommerce });
+                }
+            } catch (e) {
+                this._deinitializePollingTimer(gatewayId);
+            }
+        }, POLLING_INTERVAL);
+    }
+
+    private _reinitializeButtons({ gatewayId, methodId, paypalcommerce }: PaymentInitializeOptions) {
+        this._deinitializePollingTimer(gatewayId);
+        // @ts-ignore
+        this._paypalCommercePaymentProcessor.getButtons().close();
```

Comment:
Why do you have to call `getButtons().close()` again, when `deinitialize` already calls `paypalButtons?.close?.()`?

Reply:
fixed